### PR TITLE
Mauriciogtec/issue60

### LINKED ISSUE #60: Python API for downloading data
It is necessary to develop a Python API that enables data downloading and maintains a persistent connection for future data loaders. All download and processing operations should be conducted within a Python session, rather than a shell script.

### LINKED ISSUE #62: Add evaluator class
Add an api for evaluator

### PR DESCRIPTION
**Pull request by mauriciogtec**
Add SpaceEnv that downloads from dataverse
Fixes #60
Add evaluator class with demo for benchmarks
Fixes #62

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,4 +3,7 @@
 
 ## Added
 
-- DAPSm algorithm. Basic functionality with optimal and greedy matching.
\ No newline at end of file
+- DAPSm algorithm. Basic functionality with optimal and greedy matching.
+- Evaluator classes with demo on evaluator_demo.ipynb
+- SpaceEnv classes (same demo). Data automatically downloads given name.
+- Added Masterfile but needs to be updated with new datasets.
```
[the remark this answers is not in the document]
Comment (Naeemkh) [reply CHANGELOG.md:9]: Also 
- CausalDataset --> SpaceDataset

File: spacebench/api/dataverse.py [hunk 2]
```diff
@@ -91,80 +69,73 @@ def list_data_files(self, include_fileid=False):
...
+        Publish new dataset.
 
         Args:
             token (str): Dataverse API Token.
         """
         api = NativeApi(self.base_url, token)
         resp = api.publish_dataset(self.pid, release_type="major")
         if resp.json()["status"] == "OK":
-           LOGGER.info("Dataset published.")
+            LOGGER.info("Dataset published.")
```
Comment (Naeemkh) [inline spacebench/api/dataverse.py:115]: The log file could be more informative. This particular line, along with several others, could be enhanced with additional information. Please consider including the name of the dataset for better clarity and usefulness.

File: spacebench/api/dataverse.py [hunk 3]
```diff
@@ -194,13 +164,11 @@ def replace(self, file, file_id, token):
...
 
         json_str = json.dumps(json_dict)
-        resp = api.replace_datafile(
-            file_id, file, json_str, is_filepid=False)
+        resp = api.replace_datafile(file_id, file, json_str, is_filepid=False)
 
         if resp.json()["status"] == "ERROR":
-            LOGGER.error(
-                f"An error at replacing the file: {resp.content}")
+            LOGGER.error(f"An error at replacing the file: {resp.content}")
```
[the remark this answers is not in the document]
Comment (mauriciogtec) [reply spacebench/api/dataverse.py:174]: Same as above. The dataverse API should be updated on a separate issue, as much has become obsolete.
The fixed version should raise errors.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mauriciogtec) [reply CHANGELOG.md:9]: Done.

**Pull request merged**